### PR TITLE
[MI-2626]: Improved the UI for GroupedPartOutputControl type response from VA

### PR DESCRIPTION
**Pull request by Nityanand13**

### CODE REVIEW

File: server/plugin/virtual_agent.go [hunk 1]
```diff
@@ -407,11 +404,29 @@ func (p *Plugin) CreateOutputCardRecordAttachment(body *OutputCardRecordData) *m
...
-		Title: fmt.Sprintf("[%s](%s)", body.Label, body.Action),
-		Text:  body.Description,
+func (p *Plugin) CreatePostForGroupedPartsOutputControl(userID string, res *GroupedPartsOutputControl) error {
+	var message strings.Builder
+	for id, value := range res.Values {
+		message.WriteString(fmt.Sprintf("[%s](%s)\n%s", value.Label, value.Action, value.Description))
+		if id != len(res.Values)-1 {
+			message.WriteString("\n\n")
+		}
 	}
```
Comment (raghavaggarwal2308) [inline server/plugin/virtual_agent.go:414]:
```suggestion
	}
	
```

File: server/plugin/virtual_agent.go [hunk 2]
```diff
@@ -407,11 +404,29 @@ func (p *Plugin) CreateOutputCardRecordAttachment(body *OutputCardRecordData) *m
...
+	post := &model.Post{
+		ChannelId: channel.Id,
+		UserId:    p.botUserID,
+		Message:   message.String(),
+	}
+	if _, err = p.API.CreatePost(post); err != nil {
+		return err
+	}
+
+	return nil
```
Comment (raghavaggarwal2308) [inline server/plugin/virtual_agent.go]: There is a function "DM" available for this, use that here!

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/plugin/virtual_agent.go [hunk 3]
```diff
@@ -407,11 +404,22 @@ func (p *Plugin) CreateOutputCardRecordAttachment(body *OutputCardRecordData) *m
...
+func (p *Plugin) CreatePostForGroupedPartsOutputControl(userID string, res *GroupedPartsOutputControl) error {
+	var message strings.Builder
+	for id, value := range res.Values {
+		message.WriteString(fmt.Sprintf("[%s](%s)\n%s", value.Label, value.Action, value.Description))
+		if id != len(res.Values)-1 {
+			message.WriteString("\n\n")
+		}
 	}
+
+	if _, err := p.dm(userID, &model.Post{
```
Comment (raghavaggarwal2308) [inline server/plugin/virtual_agent.go]: Use "p.DM"

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/virtual_agent.go [hunk 4]
```diff
@@ -407,11 +404,20 @@ func (p *Plugin) CreateOutputCardRecordAttachment(body *OutputCardRecordData) *m
 	}
 }
 
-func (p *Plugin) CreateGroupedPartsOutputControlAttachment(body GroupedPartsOutputControlValue) *model.SlackAttachment {
-	return &model.SlackAttachment{
-		Title: fmt.Sprintf("[%s](%s)", body.Label, body.Action),
-		Text:  body.Description,
+func (p *Plugin) CreatePostForGroupedPartsOutputControl(userID string, res *GroupedPartsOutputControl) error {
+	var message strings.Builder
+	for id, value := range res.Values {
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go]:
```suggestion
	for index, value := range res.Values {
```

**Review by manojmalik20 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**